src/odemis/acq/stream/_sync.py

```diff
@@ -1371,22 +1378,28 @@ def _runAcquisitionEbeam(self, future):
 
     def _integrateImages(self, n):
         """
-        Integrate/sum the last n acquired images in self._acq_data and replace by the
-        integrated image. If values overflow, clip the image. If number of images = 1,
-        no image integration necessary.
-        :param n: (int) Number of images that need to be integrated (summed).
+        Integrate/sum the images in self._acq_data. If number of images = 1, no image integration necessary.
+        If number of images > 10, integrate one image after another and replace by the integrated image.
+        It's about integrating the new acquired image in self._acq_data with the last integrated image if exists.
+        If number of images <= 10, integrate/sum the last n acquired images in self._acq_data and replace by the
+        integrated image. If values overflow, clip the image.
+        Args:
+            n: (int) Number of images that need to be integrated (summed).
         """
+
         # check if we need to integrate images
         if n == 1:
             return
 
+        n = 2 if n > 10 else n  # if n > 10, then sum one image after another (n = 2, number of images to be integrated)
+
         for stream_idx, das in enumerate(self._acq_data):
             # get the metadata from last image (should be the same for all images)
             md = das[-1].metadata
             det_type = md.get(model.MD_DET_TYPE, model.MD_DT_INTEGRATING)
             # get the images, which need to be averaged/integrated
             data2process = das[-n:]
-            orig_dtype = data2process[0].dtype
+            orig_dtype = data2process[-1].dtype if n == 2 else data2process[0].dtype
 
             if det_type == model.MD_DT_NORMAL:  # SEM
```

Comment:
Average is not going to work as expected if you do it one image after one image (because the latest images would have a bigger weight than the first ones). So you need to always sum, and only after the end of the acquisition, if the detector is DT_NORMAL, then divide and restore the original dtype.

Reply:
@VictoriaDel I think you could do the division part in the method assemble_final_data if you saved to sum in live_data. (Though I haven't read the specifications for this PR/do not know all the details). In live_data you want to also save the latests SEM image to be displayed on screen.